cmake/modules/CheckDependentLibraries.cmake

```diff
@@ -179,12 +179,12 @@ endif()
 
 if(WITH_POSTGRES)
   find_package(PostgreSQL)
-  if(POSTGRESQL_FOUND)
+  if(PostgreSQL_FOUND)
```

Comment:
For the find_package above, I remember reading something in the CMake release notes some not too long time ago about PostgreSQL. 

Ok I found it: https://cmake.org/cmake/help/latest/module/FindPostgreSQL.html

since 3.14, it is included, and offers it through an imported target, the ideal way it should be used.

I see that we (for now) specify 3.11 as the minimum.

one day https://github.com/nlohmann/cmake_min_version could be used to do a binary search of what minimum version works to compile.

At the end we could argue to see what best version to use, as some nice, useful features exist in newer. 3.28 has a pretty major change in windows path search, and has nice editor integration (useful to develop with).

Reply:
There are more from cmake 3.29.1 (what I use):
* FindFontconfig
* FindGDAL
* FindIconv
* FindPostgreSQL
* FindSQLite3

Reply:
I believe we could easily bump that requirement to CMake 3.16. There will be some time before cmake build completely replaces  the current build system.

CMake version on some platforms:
`Debian 11` 3.18
`Fedora 38` 3.26
`Ubuntu 22.04` 3.22

Some other projects' requirements:
`GDAL` 3.16...3.28
`PDAL` 3.13
`PROJ` 3.16
`QGIS` 3.12

With version 3.16, also the following platforms are supported:
`Fedora 32`
`Ubuntu 20.04`

Reply:
Slackware 15 currently has 3.21. CMake 3.16 was released 5 years ago; I don't think it's unreasonable to require it.

Reply:
Even if it isn't included with the system, as it is self contained, one could download or pip install cmake easily.

On another topic, I've later read that CMake now tends to prefer using the config mode of find_package, that is, a .cmake file provided by the projects with the real info and targets needed instead of heuristics-based find modules. They do not plan to add new find modules, and they are easily out of date. Package-config files are also automatically supported by cmake. Creating the cmake config file from a project that is using cmake is done with the CMakePackageConfigHelpers  module https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#module:CMakePackageConfigHelpers
See https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#creating-packages

Reply:
> On another topic, I've later read that CMake now tends to prefer using the config mode of find_package, that is, a .cmake file provided by the projects with the real info and targets needed instead of heuristics-based find modules. 

This only works for such packages that supports it, I know PDAL does.

Reply:
Generationally, cmake 3.21 seems quite a step up from others. VS2022 (now used by osgeo4w), Msys environment support, and is more robust.

Reply:
We also have the chance that we don't have legacy compatibility to bother with, it's a new implementation.